imports/ui/components/playlists/playlist_page.js

```diff
@@ -41,11 +45,32 @@ Template.playlistPage.helpers({
   },
   songs: () => Session.get("currentPlaylist"),
   showName: () => Shows.findOne({ showId: Playlists.findOne().showId }).showName,
-  showDate: () => moment(Playlists.findOne().showDate).tz("US/Hawaii").format("LL"),
+  showNameFromId: (id) => Shows.findOne({ showId: id }).showName,
+  showDateOfCurrent: () => moment(Playlists.findOne().showDate).tz("US/Hawaii").format("LL"),
   showSlug: () => Shows.findOne({ showId: Playlists.findOne().showId }).slug,
+  showTimeFromId: (id) => Shows.findOne({showId: id}).startHour,
+  showEndFromId: (id) => Shows.findOne({showId: id}).endHour,
   showImage: () => {
     var show = Shows.findOne({ showId: Playlists.findOne().showId });
     return (show === undefined) ? false : show.featuredImage.url;
   },
-  timeBeautify: (time) => moment(time.substring(0, time.length - 3), 'HH:mm').format('hh:mm a')
+  timeFromHours: (h1, h2) => moment(h1, "HH").format('h') + "-" + moment(h2, "HH").format('hA'),
+  timeBeautify: (time) => moment(time.substring(0, time.length - 3), 'HH:mm').format('hh:mm a'),
+  dateFormat: (date) => moment(date).format("ddd. MMMM DD, YYYY"),
+  getSidebarData: () => {
+    var d = Playlists.find({}, {sort: {showDate: -1, spinPlaylistId: -1}, limit: 12}).fetch();
+    var pluckDate = (array) => _.pluck(array, "showDate");
+    var dateEq = (d1) => +d1;
+    var uniqDate = (array) => _.uniq(array, true, dateEq);
+    var plucked = pluckDate(d);
+    var dates = uniqDate(plucked);
+    var a = [];
+    for (var p = 0; p < dates.length; p++) {
+      var r = {};
```

Comment:
'a' for 'array' i can understand, but what does the 'r' stand for here? and (now i'm really splitting hairs but) why index with 'p' instead of the usual 'i'?